src/MergeContext.js

```diff
@@ -108,6 +108,87 @@ class Approval {
     }
 }
 
+class StatusCheck
+{
+    constructor(context, state, targetUrl, description) {
+        assert(context);
+        assert(state);
+        assert(targetUrl);
+        assert(description);
+
+        this.context = context;
+        this.state = state;
+        this.targetUrl = targetUrl;
+        this.description = description;
+    }
+}
+
+// Passed required status checks analysis
+class StatusResult
+{
+    // checksNumber:
+    //   for staged commits: the bot-configured number of required checks (Config.stagingChecks()),
+    //   for PR commits: GitHub configured number of required checks (requested from GitHub)
+    // context: either "PR" or "Staging";
+    constructor(checksNumber, context) {
+        assert(checksNumber !== undefined && checksNumber !== null);
+        assert(context);
+        this.checksNumber = checksNumber;
+        this.context = context;
+        this.requiredChecks = [];
+    }
+
+    addRequiredStatus(requiredCheck) {
+        assert(requiredCheck);
+        this.requiredChecks.push(requiredCheck);
+    }
+
+    approvalPending() {
```

Comment:
This method is needed to distinguish 2 kinds of pending checks: suspended and delayed. For now, the only delayed check is 'approval' check. For simplicity, I created a dedicated method (named specifically), however, probably there should be a general-purpose method with a parameter (a check name).

Reply:
Please note that my primary concern here is that the method code seems to be relying on the state of the "PR Approval" status check on GitHub. We should not rely on that. When needed, we should rely on our own, fresh calculation of the PR approval state (approved, suspended, etc.). If that concern is addressed by avoiding searching status checks for Config.approvalContext, then the method will become out of place -- it would no longer belong to the class dedicated to the analysis of those status checks because it would not be using those status checks anymore.